File: src/main/java/tk/hiddenname/smarthome/entity/Area.java
package tk.hiddenname.smarthome.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@Data
@Entity
@Table(name = "area")
@EqualsAndHashCode(of = {"id"})
@AllArgsConstructor
@NoArgsConstructor
public class Area {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(updatable = false, nullable = false)
    private Integer id;

    @NotNull
    @Size(min = 3, max = 25)
    @Column(nullable = false, length = 25)
    private String name;

    @NotNull
    @Size(min = 3, max = 50)
    @Column(nullable = false, length = 50)
    private String description;
}

File: src/main/java/tk/hiddenname/smarthome/model/Area.kt
package tk.hiddenname.smarthome.model

import com.fasterxml.jackson.annotation.JsonPropertyOrder
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table
import javax.validation.constraints.NotBlank
import javax.validation.constraints.Size

@Entity
@Table(name = "area")
@JsonPropertyOrder(value = ["id", "name", "description"])
data class Area(
        @field:NotBlank(message = "Blank")
        @field:Size(min = 3, max = 25)
        @Column(nullable = false, length = 25)
        var name: String = "",

        @field:Size(max = 50)
        @Column(nullable = false, length = 50)
        var description: String = ""
) : AbstractJpaPersistable()